bool desire_state
---
bool ack_state